services/venice-server/src/main/java/com/linkedin/venice/listener/OutboundHttpWrapperHandler.java

```diff
@@ -108,6 +109,20 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
           contentType = HttpConstants.TEXT_PLAIN;
           responseStatus = INTERNAL_SERVER_ERROR;
         }
+      } else if (msg instanceof MetadataResponse) {
+        MetadataResponse metadataResponse = (MetadataResponse) msg;
+        if (!metadataResponse.isError()) {
+          body = metadataResponse.getResponseBody();
+          schemaIdHeader = metadataResponse.getResponseSchemaIdHeader();
+        } else {
+          String errorMessage = metadataResponse.getMessage();
+          if (errorMessage == null) {
```

Comment:
Is there a case where the errorMessage can be `null`?

Reply:
There shouldn't be a case with a `null` errorMessage. I added this check because I saw the same pattern above it with the AdminResponse. That also didn't have any case where isError() is `true` without setting the errorMessage. Should it be removed?

Reply:
I think we can remove it. If we want to hide/manipulate the error msg, it's better to do it inside of the `getMetadata` method IMO.

Reply:
Got it, will remove that logic

Reply:
I think the intent in this case is to be cautious since we are receiving something on the wire coming from another process, and there is no guarantee that the other process runs on the same version of the code as this one. Looking at the code in the repo is not enough to guarantee that it will never contain null, unless we could guarantee that all versions of the software will behave as such. I'm not 100% sure this comes from the wire, I haven't looked in depth. So please assess accordingly. Thanks.